app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -149,6 +149,7 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
     override fun onStop() {
         super.onStop()
 
+        serviceLocator!!.sessionRepo.exitFullScreenIfPossible()
         sessionFeature?.stop()
```

Comment:
Did you verify that this works consistently?  I tried something similar for another problem that caused a race condition and would fail around 20% of the time.

Reply:
I tested this on IMDB and didn't see any problems, either with the un-fullscreen or the grey screen coming back.